test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java

```diff
@@ -148,4 +157,30 @@
         assertEquals(new Cause.UpstreamCause((Run) b), downstream.getLastBuild().getCause(Cause.UpstreamCause.class));
     }
 
+    @Issue("JENKINS-29876")
+    @Test
+    public void nullJobInTriggerNotCausesNPE() throws Exception {
+        final FreeStyleProject upstreamJob = r.createFreeStyleProject("upstream");
+
+        //job with trigger.job == null
+        final FreeStyleProject downstreamJob1 = r.createFreeStyleProject("downstream1");
+        final ReverseBuildTrigger reverseBuildTrigger = new ReverseBuildTrigger("upstream", Result.SUCCESS);
+        downstreamJob1.addTrigger(reverseBuildTrigger);
```

Comment:
No, this confused me during tests implementation. I thought that's why you did roundtrips in other test cases :)
see https://github.com/jenkinsci/jenkins/pull/1842#issuecomment-144721166